src/main/java/org/passay/rule/DigestHistoryRule.java

```diff
@@ -57,10 +59,17 @@ public DigestHistoryRule(final HashBean<String> bean, final Charset set)
    * @return  true if passwords match, false otherwise.
    */
   @Override
-  protected boolean matches(final String password, final PasswordData.Reference reference)
+  protected boolean matches(final UnicodeString password, final PasswordData.Reference reference)
   {
     final PasswordData.Salt salt = reference.getSalt();
-    final String undigested = salt == null ? password : salt.applyTo(password);
-    return hashBean.compare(reference.getPassword(), undigested.getBytes(charset));
+    final CharBuffer buffer = password.toCharBuffer();
+    final CharBuffer undigested = salt == null ? buffer : salt.applyTo(buffer);
+    try {
+      return hashBean.compare(
```

Comment:
Sure would be nice if you could use the `CharBuffer` directly rather than having to copy to a byte array. Maybe there's a future enhancement to Cryptacular that would support this.